### ISSUE TITLE
xgb.train parameters default value settings and overfitting issues

### ISSUE THREAD

**Issue by yilinwu123** [dmlc/xgboost#7415]
Dear all:

I am using the package 'xgboost' in R to do research. While using the package, I noticed that the main function ‘xgb.train’ has several parameters with default values specified. To tailor these parameters to the datasets at hand, I’m hoping to better understand how to bestselect their values. If I may ask, could you please kindly advise: 

(1) what approach and metric were used to select the default values?
(2) what are the relative importance of the various parameters? There is the problem of overfitting for the results I got so far. I am tuning parameters and wondering which paramters could help reduce overfitting more efficiently than others.

I’d also be grateful if you could offer any additional advices on setting suitable parameter values.

Thank you very much for your consideration and help.

**Comment by trivialfis** [comment]
> what approach and metric were used to select the default values?

I don't know, mostly just old heuristics.

> what are the relative importance of the various parameters? 

We have better tools and algorithms for selecting these parameters now, for instance, optuna in Python.

Related issues: https://github.com/dmlc/xgboost/issues/6034 https://github.com/dmlc/xgboost/issues/4986  Right now I recommend using one of those tuning frameworks instead of relying on the default parameter set.

**Comment by yilinwu123** [comment]
Thanks for the quick response. I will look at the methods you mentioned. So far, I just used simple grid search to tune several parameters and have not found some effective results. Thanks a lot!

**Comment by trivialfis** [comment]
Excellent.  Closing now, feel free to share your results as follow-up issues/PRs. ;-)
